pkg/asset/cluster/aws/aws.go

```diff
@@ -145,18 +164,26 @@ func tagSharedIAMRoles(ctx context.Context, clusterID string, installConfig *ins
 
 	logrus.Debugf("Tagging shared instance roles: %v", sets.List(iamRoles))
 
-	session, err := installConfig.AWS.Session(ctx)
+	tagKey, tagValue := sharedTag(clusterID)
+
+	cfg, err := configv2.LoadDefaultConfig(context.TODO(), configv2.WithRegion(installConfig.Config.Platform.AWS.Region))
```

Comment:
```suggestion
	cfg, err := configv2.LoadDefaultConfig(ctx, configv2.WithRegion(installConfig.Config.Platform.AWS.Region))
```

Should we use context passed to `tagSharedIAMRoles`?